web/src/pages/billing/BillingCostByTime.tsx

```diff
@@ -335,16 +335,69 @@ const BillingCostByTime: React.FunctionComponent = () => {
         }
     }, [start, end, groupBy, selectedData])
 
+    const exportOptions = [
+        { key: 'csv', text: 'Export to CSV', value: 'csv' },
+        { key: 'tsv', text: 'Export to TSV', value: 'tsv' },
+    ]
+
+    const exportToFile = (format: 'csv' | 'tsv') => {
+        // Create header: Date plus group/column names
+        const headerData = ['Date', ...groups]
+
+        // Build rows: for each object in `data` (each date)
+        const rowData = data.map((row) => {
+            // Each `row` is { date, values }
+            const dateStr =
+                row.date instanceof Date ? row.date.toISOString().slice(0, 10) : String(row.date)
+
+            // Get values for each group, blank if not found
+            const vals = groups.map((group) =>
+                row.values[group] !== undefined && row.values[group] !== null
```

Comment:
rather than checking for what the value shouldn't be, it is less verbose to check for what it should be: `typeof row.values[group] === 'number'`